ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -404,13 +389,32 @@ private int HandlePermEvent(ref Interop.ProjFS.Event ev)
             return Result.ENotYetImplemented;
         }
 
-        private static unsafe class NativeMethods
+        private static unsafe class NativeFileWriter
         {
-            [DllImport("libc", EntryPoint = "strlen")]
-            public static extern ulong Strlen(byte* s);
+            public static bool TryWrite(int fd, byte[] bytes, uint byteCount)
```

Comment:
> I moved the write-bytes-to-file loop down into a NativeFileWriter class, which I think parallels the NativeFileReader found in GVFS.Platform.Mac.MacFileSystem. That also tidies up the main class in VirtualizationInstance a bit.

I think that `NativeFileReader` and `NativeFileWriter` are probably being used for different things.  `NativeFileReader` is used during `gvfs prefetch` (when `--hydrate` is specified) to force files to be hydrated.

I am guessing that `NativeFileWriter` is used to actually do the hydration in the GetFileStream callback?  If that's the case it looks like the parallel to [PrjFS_WriteFileContents ](https://github.com/Microsoft/VFSForGit/blob/e3a7dee707ac847e9ab169a87c632829e711af7c/ProjFS.Mac/PrjFSLib/PrjFSLib.cpp#L589).

(I don't think any change is needed here, I just wanted to make sure we're all on the same page).

Reply:
Indeed, that's all correct -- I apologize, I didn't mean to imply a functional parallel with the `NativeFileReader`, only that the implementation (i.e., a `NativeFile*` helper class) was similar.

The macOS `PrjFS_WriteFileContents()` can just call `fwrite()` natively in C++, but we do a bit more work in the Linux `GetFileStream` callback because we're in C#.